jekyll/_cci2/nomad-metrics.md

```diff
@@ -0,0 +1,225 @@
+    servers = ["$SERVICES_PRIVATE_IP:4647"]
+    node_class = "linux-64bit"
+    options = {"driver.raw_exec.enable" = "1"}
+}
+
+telemetry {
+    publish_node_metrics = true
+    statsd_address = "$SERVICES_PRIVATE_IP:8125"
+}
+EOT
+
+echo "--------------------------------------"
+echo "      Creating nomad.conf"
+echo "--------------------------------------"
+cat <<EOT > /etc/init/nomad.conf
+start on filesystem or runlevel [2345]
+stop on shutdown
+
+script
+    exec nomad agent -config /etc/nomad/config.hcl
+end script
+EOT
+
+echo "--------------------------------------"
+echo "   Creating ci-privileged network"
+echo "--------------------------------------"
+docker network create --driver=bridge --opt com.docker.network.bridge.name=ci-privileged ci-privileged
+
+echo "--------------------------------------"
+echo "      Starting Nomad service"
+echo "--------------------------------------"
+service nomad restart
+
+echo "--------------------------------------"
+echo "      Setting up Nomad metrics"
+echo "--------------------------------------"
+docker pull $CONTAINER_IMAGE
+docker rm -f $CONTAINER_NAME || true
+
+# Not using --detach so that upstart can perform log management and process
+# monitoring
+docker run -d --name $CONTAINER_NAME \
+    --rm \
+    --net=host \
+    --userns=host \
+    $CONTAINER_IMAGE \
+    start --nomad-uri=http://localhost:4646 --statsd-host=$SERVICES_PRIVATE_IP --statsd-port=$NOMAD_METRICS_PORT --client
+
+```
+
+#### Updating the Auto Scaling Group
+
+1. Select the `Auto Scaling Groups` (ASG) link in the the sidebar on the left.
+2. Select the ASG with a name tag similar to`*_nomad_clients_asg`
+3. In the description box at the bottom, select the `Edit` button
+4. Select the newly created Launch Configuration from the drop-down.
+5. Press the `Save` button.
+6. You will need to terminate the existing Nomad Client instances in order for newer instances with the new Launch Configuration to be created.  Ideally, this should only be done during off-hours or inside of a maintenance window as it could prematurely terminate running jobs.
+
+
+
+## StatsD Metrics
+
+## --server
+
+Name                                      | Type  | Description
+------------------------------------------|-------|-------------
+`circle.nomad.server_agent.poll_failure`  | Gauge | 1 if the last poll of the Nomad agent failed; 0 otherwise.  This gauge is set independent of `circle.nomad.client_agent.poll_failure` when nomad-metrics is operating in `--client` and `--server` modes simultaneously.
+`circle.nomad.server_agent.jobs.pending`  | Gauge | Total number of pending jobs across the cluster.
+`circle.nomad.server_agent.jobs.running`  | Gauge | Total number of running jobs across the cluster.
+`circle.nomad.server_agent.jobs.complete` | Gauge | Total number of complete jobs across the cluster.
+`circle.nomad.server_agent.jobs.dead`     | Gauge | Total number of dead jobs across the cluster.
+{: class="table table-striped"}
```

Comment:
I discovered 6 other metrics related to `poll_time`. Not sure if they should be documented in this section.

metric_type | Metric Name
----- | -----
timing | circle.nomad.server_agent.poll_time_lower
timing | circle.nomad.server_agent.poll_time_mean
timing | circle.nomad.server_agent.poll_time_stddev
timing | circle.nomad.server_agent.poll_time_count
timing | circle.nomad.server_agent.poll_time_sum
timing | circle.nomad.server_agent.poll_time_upper